src/common/meta/src/key/flow.rs

```diff
@@ -230,6 +230,99 @@ impl FlowMetadataManager {
         Ok(())
     }
 
+    /// Update metadata for flow and returns an error if old metadata IS NOT exists.
+    pub async fn update_flow_metadata(
+        &self,
+        flow_id: FlowId,
+        flow_info: FlowInfoValue,
+        prev_flow_info: FlowInfoValue,
+        flow_routes: Vec<(FlowPartitionId, FlowRouteValue)>,
+    ) -> Result<()> {
```

Comment:
We need to use the `DeserializedValueWithBytes`

Here is an example:
https://github.com/discord9/greptimedb/blob/e3c9e2e3bffbdf5432a7a0210acd6608a98fada5/src/common/meta/src/key/table_info.rs#L171-L176